pyradiosky/skymodel.py

```diff
@@ -2578,6 +2803,125 @@ def read_idl_catalog(
+            Option to check acceptable range of the values of parameters after
+            downselecting data on this object (the default is True, meaning the
+            acceptable range check will be done).
+        """
+        if run_check:
+            self.check(
+                check_extra=check_extra, run_check_acceptability=run_check_acceptability
+            )
+
+        if self.history is None:
+            self.history = self.pyradiosky_version_str
+        else:
+            if not uvutils._check_history_version(
+                self.history, self.pyradiosky_version_str
+            ):
+                self.history += self.pyradiosky_version_str
+
+        with h5py.File(hdf5_filename, "w") as fileobj:
+            # create header
+            header = fileobj.create_group("Header")
+            # write out UVParameters
+            header_params = [
+                "_Ncomponents",
+                "_Nfreqs",
+                "_component_type",
+                "_spectral_type",
+                "_ra",
+                "_dec",
+                "_history",
+                "_name",
+                "_nside",
+                "_hpx_inds",
+                "_freq_array",
+                "_reference_frequency",
+                "_spectral_index",
+                "_beam_amp",
+                "_extended_model_group",
+            ]
+            for par in header_params:
+                param = getattr(self, par)
+                val = param.value
+                parname = param.name
+
+                # Skip if parameter is unset.
+                if val is None:
+                    continue
+
+                # Extra attributes for astropy Quantity-derived classes.
+                unit = None
+                angtype = None
+                if isinstance(val, units.Quantity):
+                    if isinstance(val, Latitude):
+                        angtype = "latitude"
+                    elif isinstance(val, Longitude):
+                        angtype = "longitude"
+                    unit = val.unit.name
+                    val = val.value
+
+                try:
+                    dtype = val.dtype
+                except AttributeError:
+                    dtype = np.dtype(type(val))
+
+                if dtype.kind == "U":
+                    if isinstance(val, (list, np.ndarray)):
+                        header[parname] = np.asarray(val, dtype="bytes")
+                    else:
+                        header[parname] = np.string_(val)
+                else:
+                    header[parname] = val
+
+                if unit is not None:
+                    header[parname].attrs["unit"] = unit
+                if angtype is not None:
+                    header[parname].attrs["angtype"] = angtype
+
+                # Strings and arrays of strings require special handling.
+
+            # write out the stokes array
+            dgrp = fileobj.create_group("Data")
+            dgrp.create_dataset(
```

Comment:
It looks like the `chunks` keyword isn't actually passed to the dataset here.

Reply:
Thanks!